docs/sources/operators-guide/running-production-environment-with-helm/configuration-with-helm.md

```diff
@@ -0,0 +1,531 @@
+```
+
+This command shows the differences between the running installation and the installation that would result from executing the `helm upgrade` command. Search for `name: mimir-config` in the output to see the difference in configuration settings. See [Example output of helm diff command](#example-output-of-helm-diff-command) for a concrete example.
+
+> **Note:** CLI flags and their difference are found in the `Deployment` and `StatefulSet` objects.
+
+## Manage the configuration externally
+
+Prepare the configuration as text. It cannot include Helm template functions or value evaluations. The configuration may include references to environment variables as explained in [Injecting credentials](#injecting-credentials).
+
+Decide whether you want to use a `ConfigMap` or `Secret` to store the configuration. Handling `ConfigMap` is a little bit simpler, but beware of sensitive information.
+
+### Use external ConfigMap
+
+Prepare a `ConfigMap` object where the configuration is placed under the `mimir.yaml` data key.
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: my-mimir-config
+data:
+  mimir.yaml: |
+    <configuration>
+```
+
+Replace `<configuration>` with the configuration as multiline text, be mindful of indentation. The name `my-mimir-config` is just an example.
+
+Set the following values in your custom values file (or on the Helm command line):
+
+```yaml
+useExternalConfig: true
+externalConfigSecretName: my-mimir-config
+externalConfigVersion: "0"
+```
+
+### Use external Secret
+
+Prepare a `Secret` object where the configuration is base64-encoded and placed under the `mimir.yaml` data key.
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: my-mimir-config
+data:
+  mimir.yaml: <configuration-base64>
+```
+
+Replace `<configuration-base64>` with the configuration encoded as base64 format string. The name `my-mimir-config` is just an example.
+
+Set the following values in your custom values file (or on the Helm command line):
+
+```yaml
+useExternalConfig: true
+externalConfigSecretName: my-mimir-config
+configStorageType: Secret
+externalConfigVersion: "0"
+```
+
+### Update the configuration
+
+In order to make components aware of configuration changes, either:
+
+- Update the value in `externalConfigVersion` and run `helm update`
+- or restart components affected by the configuration change manually
+
+## Injecting credentials
+
+Credentials should be kept in `Secret` objects or in a credential vault. The Helm chart value `global.extraEnvFrom` can be used to inject the credentials into the runtime environment variables of the Grafana Mimir components. The data keys will become environment variables and usable in the Grafana Mimir configuration. For example `AWS_SECRET_ACCESS_KEY` can be referenced as `${AWS_SECRET_ACCESS_KEY}` in the configuration. See the [Example](#example-of-configuration-managed-with-helm) for a practical application.
```

Comment:
credential vault sounds like a generic term. Do you need to mention it here? The first sentence sounds like advice. Instead we can state that the chart allows to inject secrets from a secret.

also some passive->active voice
 
```suggestion
Credentials should be kept in `Secret` objects or in a credential vault. You can use the Helm chart value `global.extraEnvFrom` to inject the credentials into the runtime environment variables of the Grafana Mimir components. The data keys will become environment variables and usable in the Grafana Mimir configuration. For example `AWS_SECRET_ACCESS_KEY` can be referenced as `${AWS_SECRET_ACCESS_KEY}` in the configuration. See the [Example](#example-of-configuration-managed-with-helm) for a practical application.
```